flink-connectors/flink-connector-aws-kinesis-data-streams/src/test/java/org/apache/flink/connectors/kinesis/testutils/KinesaliteContainer.java

```diff
@@ -99,25 +99,41 @@ public Properties getHostProperties() {
     }
 
     /** Returns the client to access the container from outside the docker network. */
-    public KinesisAsyncClient getContainerClient() throws URISyntaxException {
-        return getClient(getContainerEndpointUrl());
+    public KinesisAsyncClient getContainerClient(SdkAsyncHttpClient httpClient)
+            throws URISyntaxException {
+        return getClient(getContainerEndpointUrl(), httpClient);
     }
 
-    /** Returns the client to access the host from inside the docker network. */
+    /** Returns the client with the default async http client. */
     public KinesisAsyncClient getHostClient() throws URISyntaxException {
-        return getClient(getHostEndpointUrl());
+        return getHostClient(buildSdkAsyncHttpClient());
     }
 
-    public KinesisAsyncClient getClient(String endPoint) throws URISyntaxException {
+    /** Returns the client to access the host from inside the docker network. */
+    public KinesisAsyncClient getHostClient(SdkAsyncHttpClient httpClient)
+            throws URISyntaxException {
+        return getClient(getHostEndpointUrl(), httpClient);
+    }
+
+    public KinesisAsyncClient getClient(String endPoint, SdkAsyncHttpClient httpClient)
+            throws URISyntaxException {
         return KinesisAsyncClient.builder()
                 .endpointOverride(new URI(endPoint))
                 .region(REGION)
                 .credentialsProvider(
                         () -> AwsBasicCredentials.create(getAccessKey(), getSecretKey()))
-                .httpClient(buildSdkAsyncHttpClient())
+                .httpClient(httpClient)
                 .build();
     }
 
+    public SdkAsyncHttpClient buildSdkAsyncHttpClient() {
```

Comment:
There is so much duplicated code already, why cant we reuse org.apache.flink.connector.aws.testutils.AWSServicesTestUtils#createHttpClient here?

Reply:
The defaults in `AWSGeneralUtil` has been tuned for the service client connecting to the actual KDS service. I would have to set quite a number of defaults back to the original Netty Nio defaults in order to get `KinesaliteContainer` working here.